test/libzip_examples/zipfs-stat.c

```diff
@@ -0,0 +1,77 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <zip.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/stat.h>
+
+char *add_slash(char *path)
+{
+	long int lenth = strlen(path);
+	char *path_add_slash = malloc(sizeof(char) * (lenth + 2));
+	strncpy(path_add_slash, path, lenth);
+	path_add_slash[lenth] = '/';
+	path_add_slash[lenth + 1] = '\0';
+	return path_add_slash;
+}
+
+void show_stat_zip(struct stat *st)
+{
+	printf("mode : %o\n", st->st_mode);
+	printf("size : %ld\n", st->st_size);
+}
+
+void show_stat_file(zip_stat_t *st)
+{
+	printf("size : %ld\n", st->size);
+}
+
+int main(int argc, char *argv[])
+{
+	char *full_path = argv[argc - 1];
+	char buf[50] = { 0 };
+	char *delim = "/";
+	//char *name[50];
+	char *fname = NULL;
+	char *archive_name = NULL;
+	zip_t *archive = NULL;
+	zip_stat_t file_stat;
+	zip_stat_init(&file_stat);
+	struct stat stat_buff;
+	int err = 0, count = 0;
+
+	strcpy(buf, full_path);
+	archive_name = strtok(buf, delim);
+	long lenth = strlen(archive_name);
+	fname = full_path + lenth;
+
+	archive = zip_open(archive_name, ZIP_CHECKCONS, &err);
+	if (archive == NULL) {
+		printf("error!\n");

```

Comment:
The user will appreciate it to see an informative error message.

```suggestion
		printf("error: Failed to open archive %s\n", archive_name);

```